tests/unit/developers-dao.test.js

```diff
@@ -0,0 +1,167 @@
+const appRoot = require('app-root-path');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const chaiExclude = require('chai-exclude');
+const config = require('config');
+const _ = require('lodash');
+const sinon = require('sinon');
+
+sinon.replace(config, 'get', () => ({ oracledb: {} }));
+const conn = appRoot.require('api/v1/db/oracledb/connection');
+const developersDao = appRoot.require('api/v1/db/oracledb/developers-dao');
+const developersSerializer = appRoot.require('api/v1/serializers/developers-serializer');
+
+chai.should();
+chai.use(chaiExclude);
+chai.use(chaiAsPromised);
+
+describe('Test developers-dao', () => {
+  const fakeId = 'fakeId';
+
+  beforeEach(() => {
+    sinon.stub(conn, 'getConnection').resolves({
+      execute: (sql, sqlParams) => {
+        const sqlResults = {
+          multiResults: {
+            rows: [{}, {}],
+            outBinds: {
+              outId: [1],
+            },
+          },
+          singleResult: {
+            rows: [{}],
+            outBinds: {
+              outId: [1],
+            },
+          },
+          emptyResult: { rows: [] },
+        };
+        let result;
+        if (sql.includes('DELETE')) {
+          result = sqlResults.emptyResult;
+        } else if ('developerId' in sqlParams || 'id' in sqlParams) {
+          result = sqlResults.singleResult;
+        } else {
+          result = sqlResults.multiResults;
+        }
+        // return 'developerId' in sqlParams ? sqlResults.singleResult : sqlResults.multiResults;
+        return result;
+      },
+      close: () => null,
+    });
+  });
+  afterEach(() => sinon.restore());
+
+  it('getDevelopers should return multiResult', () => {
+    const developersSerializerStub = sinon.stub(developersSerializer, 'serializeDevelopers');
+    developersSerializerStub.returnsArg(0);
+
+    const expectedResult = [{}, {}];
+    const result = developersDao.getDevelopers(fakeId);
+    return result.should
+      .eventually.be.fulfilled
+      .and.deep.equal(expectedResult)
+      .then(() => {
+        sinon.assert.callCount(developersSerializerStub, 1);
+      });
+  });
+
+  it('getDeveloperById should return singleResult', () => {
+    const developersSerializerStub = sinon.stub(developersSerializer, 'serializeDeveloper');
+    developersSerializerStub.returnsArg(0);
+
+    const fulfilledCases = [
+      { expectedResult: {} },
+    ];
+
+    const fulfilledPromises = [];
+    _.each(fulfilledCases, ({ expectedResult }) => {
+      const result = developersDao.getDeveloperById(
+        fakeId,
+      );
+      fulfilledPromises.push(result.should
+        .eventually.be.fulfilled
+        .and.deep.equal(expectedResult)
+        .then(() => {
+          sinon.assert.callCount(developersSerializerStub, fulfilledCases.length);
+        }));
+    });
+    return Promise.all(fulfilledPromises);
```

Comment:
You only tested the normal case for all of your test cases, but your test case should also cover the case if the SQL query returns bad data that you didn't expect. You need to have your DAO handle them properly and make sure your unit test does test the error handling logic. For example: https://github.com/osu-mist/terms-api/blob/master/tests/unit/terms-dao.test.js#L44-L61
Same comment with all of the other test cases.